lib/assert.js

```diff
@@ -91,6 +91,36 @@ export function isArrayOfClass(object, expectedClass) {
   }
 }
 
+export function isInterface(object, expectedProperties) {
+  const parameter = Object.keys(object).pop();
+  const message = `The '${parameter}' parameter should be an object that has
+    '${expectedProperties}' properties.`;
+
+  for (let property of expectedProperties) {
+    if (!(property in object[parameter])) {
+      throwError(message);
+    }
+  }
+}
+
+export function isArrayOfInterface(object, expectedProperties) {
+  const parameter = Object.keys(object).pop();
+  const message = `The '${parameter}' parameter should be an array containing
+    objects that each have '${expectedProperties}' properties.`;
+
+  if (!Array.isArray(object[parameter])) {
```

Comment:
not 100% of this check, the function name 'isArrayOfInterface' suggests that object should be an array and this asserts each entry in the array has an explicit interface.